solidity/test/bridge/Bridge.test.ts

```diff
@@ -2233,229 +2252,1840 @@ describe("Bridge", () => {
+                          // Treasury balance should be increased by the total
+                          // treasury fee for all requests. However, only
+                          // pending requests are taken into account and all
+                          // reported timeouts should be ignored because the
+                          // appropriate bookkeeping was already made upon
+                          // timeout reports. See docs of the used test data
+                          // for details.
+                          expect(
+                            treasuryBalance.afterProof.sub(
+                              treasuryBalance.beforeProof
+                            )
+                          ).to.equal(2217)
+                        })
+
+                        it("should not change redeemers balances in any way", async () => {
+                          // Redemption proof submission should NEVER cause
+                          // any change of the redeemers balances.
+                          for (
+                            let i = 0;
+                            i < data.redemptionRequests.length;
+                            i++
+                          ) {
+                            const redeemerBalance = redeemersBalances[i]
+
+                            expect(
+                              redeemerBalance.afterProof.sub(
+                                redeemerBalance.beforeProof
+                              )
+                            ).to.be.equal(
+                              0,
+                              `Balance of redeemer with index ${i} has changed`
+                            )
+                          }
+                        })
+                      }
+                    )
+
+                    context(
+                      "when output vector contains a pending requested redemption with wrong amount",
+                      () => {
+                        const data: RedemptionTestData = JSON.parse(
+                          JSON.stringify(MultiplePendingRequestedRedemptions)
+                        )
+
+                        let outcome: Promise<RedemptionScenarioOutcome>
+
+                        before(async () => {
+                          await createSnapshot()
+
+                          // Alter the last redemption requests in the test
+                          // data and set such an amount that will cause
+                          // the Bitcoin redemption transaction to be deemed
+                          // as invalid due to a wrong amount. The corresponding
+                          // transaction output has the value of 191169 so to
+                          // make this test scenario happen, the request amount
+                          // must be way different (lesser or greater) than the
+                          // output value.
+                          data.redemptionRequests[4].amount = 100000
```

Comment:
Just like in the cost of a single output, I think we could improve tests by adding two separate cases:
- requested amount - treasury fee - max Bitcoin transaction fee - 1
- requested amount + 1

We would then have all corner cases covered.

Reply:
And same response as in https://github.com/keep-network/tbtc-v2/pull/155#discussion_r821661775